python/pyarrow/pandas-shim.pxi

```diff
@@ -183,6 +187,12 @@ cdef class _PandasAPIShim(object):
         else:
             return False
 
+    cpdef is_extension_array_dtype(self, obj):
+        if self._is_extension_array_dtype:
```

Comment:
This also needs a `self._check_import()` or `self._have_pandas_internal()` I think, in case this is the first function called